src/NLog/Targets/MailTarget.cs

```diff
@@ -408,10 +423,10 @@ private void ConfigureMailClient(LogEventInfo lastEvent, ISmtpClient client)
 
         private void CheckRequiredParameters()
         {
-            if (!this.UseSystemNetMailSettings && this.SmtpServer == null)
+            if (!this.UseSystemNetMailSettings && this.SmtpServer == null && string.IsNullOrEmpty(this.PickupDirectoryLocation))
```

Comment:
Not sure if this is correct, if `UseSystemNetMailSettings` is `true` then it's OK to have `SmtpServer` and `PickupDirectoryLocation` `null`.